include/aws/s3/private/s3_util.h

```diff
@@ -108,11 +111,38 @@ AWS_S3_API
 void replace_quote_entities(struct aws_allocator *allocator, struct aws_string *str, struct aws_byte_buf *out_buf);
 
 /* TODO could be moved to aws-c-common. */
+AWS_S3_API
 int aws_last_error_or_unknown(void);
 
 AWS_S3_API
 void aws_s3_add_user_agent_header(struct aws_allocator *allocator, struct aws_http_message *message);
 
+AWS_S3_API
+int aws_s3_parse_content_range_response_header(
```

Comment:
I'll comment it out that `out_object_size` is required and `out_range_start/out_range_end` are optional